### PR TITLE
[ARTS-663] init service uses spring config / config-server

### PR DESCRIPTION
**Pull request by tamirkamara**
## Description
Please replace this text with a summary of the change and which issue is fixed.

### Changes
- [ARTS-663] - some description

### Checklist:

- [x] Branch name follows convention (feature/arts-#-short-name OR fix/arts-#-short-name)
- [x] I have included a short-meaningful title, description and changes for this PR

[ARTS-663]: https://ndph-arts.atlassian.net/browse/ARTS-663

### CODE REVIEW

File: init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/loader/Loader.java [hunk 1]
```diff
@@ -29,13 +31,19 @@ public Loader(Trial trialConfig, PractitionerServiceInvoker practitionerServiceI
         this.siteServiceInvoker = siteServiceInvoker;
     }
 
-
     @Override
     public void run(String... args) {
+        // TODO: remove this temporary fix in this story: https://ndph-arts.atlassian.net/browse/ARTS-362
+        try {
+            // Give the other services some time to come online.
+            Thread.sleep(delayStartInSeconds * 1000);
```
Comment (guybartal) [inline init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/loader/Loader.java:39]: do we still need this after having the retry mechanism?

File: init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/service/ServiceInvoker.java [hunk 2]
```diff
@@ -39,6 +41,7 @@ protected ServiceInvoker(WebClient webClient) {
                     .body(Mono.just(payload), payload.getClass())
                     .retrieve()
                     .bodyToMono(responseExpected)
+                    .retryWhen(Retry.backoff(9, Duration.ofSeconds(5)).maxBackoff(Duration.ofSeconds(30)))
```
Comment (guybartal) [inline init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/service/ServiceInvoker.java:44]: Consider filtering by server error, not sure, but maybe other errors like 4xx might also get included.

    .retryWhen(Retry.backoff(9, Duration.ofSeconds(5)).maxBackoff(Duration.ofSeconds(30)))
            .filter(this::is5xxServerError));

//...

private boolean is5xxServerError(Throwable throwable) {
    return throwable instanceof WebClientResponseException &&
            ((WebClientResponseException) throwable).getStatusCode().is5xxServerError();
}

File: init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/loader/Loader.java (hunk 1 above)
Comment (tamirkamara) [reply init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/loader/Loader.java:39]: I thought it's a short way to save some retries and errors in the log. And anyway, we can remove it if/when we're confidant Terraform brings this service up at the appropriate time (hence the TODO comment)

**Review by guybartal — APPROVED** [review]

File: init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/service/ServiceInvoker.java [hunk 3]
```diff
@@ -39,6 +47,8 @@ protected ServiceInvoker(WebClient webClient) {
                     .body(Mono.just(payload), payload.getClass())
                     .retrieve()
                     .bodyToMono(responseExpected)
+                    .retryWhen(Retry.backoff(maxWebClientAttempts,
```
Comment (shajidissa) [inline init-service/src/main/java/uk/ac/ox/ndph/mts/init_service/service/ServiceInvoker.java:50]: maybe we can also log how many retries it takes

 .retryWhen(Retry.backoff(maxWebClientAttempts,
                            Duration.ofSeconds(5)).maxBackoff(Duration.ofSeconds(30)).doAfterRetry(retrySignal -> {
                                LOGGER.info("Retried " + retrySignal.totalRetries());
                            }))

**Review by shajidissa — APPROVED** [review]
just a minor logging comment.

**Pull request merged**